app/integration_action.go

```diff
@@ -250,6 +250,75 @@ func (a *App) DoActionRequest(rawURL string, body []byte) (*http.Response, *mode
 	return resp, nil
 }
 
+type LocalResponseWriter struct {
+	data   []byte
+	status int
+}
+
+func (w *LocalResponseWriter) Header() http.Header {
```

Comment:
What about the headers? I have an implementation in my PR you could look at: https://github.com/mattermost/mattermost-server/pull/12829/files#diff-80d2a5d8acd634624ac04f5c6365a550R803

Reply:
I followed your approach with the headers, thank you.